I tried to install this from your branch on my Mac with llvm-clang on Monterey and got the following error:
```
/Users/heinzell/work/jedi-stack/spack-stack-new-joint/spack-stack-develop-20220215/spack/lib/spack/env/clang/gfortran   splat.f90
Undefined symbols for architecture x86_64:
  "_main", referenced from:
     implicit entry/start for main executable
ld: symbol(s) not found for architecture x86_64
collect2: error: ld returned 1 exit status
make[1]: *** [splat.o] Error 1
```
Too late to figure this out today. Could be one of these pseudo case-sensitive filesystem issues.

--- a/configs/repos/ufs/packages/wgrib2/package.py
+++ b/configs/repos/ufs/packages/wgrib2/package.py
@@ -94,6 +94,16 @@ class Wgrib2(MakefilePackage):
 
     # Disable parallel build
     parallel = False
+    # Use Spack compiler wrapper flags
+
+    def inject_flags(self, name, flags):
+        if name == 'cflags':
+            if self.spec.compiler.name == 'apple-clang':
+                flags.append('-Wno-error=implicit-function-declaration')
+                return (flags, None, None)
+        return (flags, None, None)
+
+    flag_handler = inject_flags
 
     def url_for_version(self, version):
         url = "https://www.ftp.cpc.ncep.noaa.gov/wd51we/wgrib2/wgrib2.tgz.v{}"
@@ -106,6 +116,15 @@ class Wgrib2(MakefilePackage):
             value = int(spec.variants[variant_name].value)
             makefile.filter(r'^%s=.*' % makefile_option,
                             '{}={}'.format(makefile_option, value))
+
+    def setup_build_environment(self, env):
+        comp_sys = ''
+        if self.spec.compiler.name == 'intel':
+            comp_sys = 'intel_linux'
+        elif self.spec.compiler.name == 'gcc':
+            comp_sys = 'gnu_linux'
+
+        env.set('COMP_SYS', comp_sys)
 
     def build(self, spec, prefix):
         make()

